UnityProject/Assets/Scripts/HealthV2/Living/Damage/Trauma/TraumaLogic.cs

```diff
@@ -46,6 +46,13 @@ public virtual void OnTakeDamage(BodyPartDamageData data)
 		/// </summary>
 		protected void GenericStageProgression()
 		{
+			if (gameObject == null || bodyPart == null)
+			{
+				Logger.LogWarning(
```

Comment:
this should be errors since it Shouldn't be happening in the first place

Reply:
no, this happens **after** the second stage that turns these into null.

Reply:
shouldn't it be removed From the update if it's null

Reply:
its an event, not an update.
It will go through the entire list regardless of what happens, so we tell it that if "hey, these two are missing or the stage is missing; assume that something has changed and return."